op-node/node/bvm_api_helper.go

```diff
@@ -85,31 +89,62 @@ func getBTCFinalityForBlockNum(ctx context.Context, blockNum uint64, driver driv
 
 	log.Trace("going to query for keystone", "keystone", spew.Sdump(l2Keystone), "nextKeystone", spew.Sdump(nextKeystone), "prevKeystoneHash", hex.EncodeToString(prevKeystoneHash[:]))
 
-	l2KeystonesToQuery := make([]hemi.L2Keystone, 1)
-	l2KeystonesToQuery[0] = *l2Keystone
+	client := &http.Client{}
+	kssHash := hemi.L2KeystoneAbbreviate(*l2Keystone).Hash()
+	u := fmt.Sprintf("%v/v%v/keystonefinality/%v",
+		bfgURL, bfgapi.APIVersion, kssHash)
+	req, err := http.NewRequestWithContext(ctx, http.MethodGet, u, nil)
+	if err != nil {
+		return emptyFin, err
+	}
+
+	resp, err := client.Do(req)
```

Comment:
`http.DefaultClient` is usually preferable to creating a new empty `http.Client{}` - Also, do we want to add a timeout to this request, or is that handled by the `ctx`?
```suggestion
	kssHash := hemi.L2KeystoneAbbreviate(*l2Keystone).Hash()
	u := fmt.Sprintf("%v/v%v/keystonefinality/%v",
		bfgURL, bfgapi.APIVersion, kssHash)
	req, err := http.NewRequestWithContext(ctx, http.MethodGet, u, nil)
	if err != nil {
		return emptyFin, err
	}

	resp, err := http.DefaultClient.Do(req)
```

Reply:
hey @joshuasing yeah we can use the default client 👍 

and this is from the docs for `NewRequestWithContext` https://pkg.go.dev/net/http#NewRequestWithContext:
> the context controls the entire lifetime of a request and its response: obtaining a connection, sending the request, and reading the response headers and body.

Reply:
Thanks! Made the change